### ISSUE TITLE
Detect must-gather directory structure / subdirectories

### ISSUE THREAD

**Issue by simonkrenger** [elmiko/okd-camgi#27]
Thanks for making this tool available. On my first try using the tool I encountered the following error when specifying a `must-gather` directory:

~~~
$ okd-camgi --webbrowser workspace/cases/01234567/must-gather.local.817596368927932123/
Traceback (most recent call last):
  File "/home/skrenger/.local/bin/okd-camgi", line 8, in <module>
    sys.exit(main())
  File "/home/skrenger/.local/lib/python3.10/site-packages/okd_camgi/main.py", line 49, in main
    content = load_index_from_path(path)
  File "/home/skrenger/.local/lib/python3.10/site-packages/okd_camgi/main.py", line 27, in load_index_from_path
    index_context = IndexContext(mustgather)
  File "/home/skrenger/.local/lib/python3.10/site-packages/okd_camgi/contexts.py", line 274, in __init__
    mapipods = sorted([PodContext(pod) for pod in mustgather.pods('openshift-machine-api')], key=lambda p: p['metadata']['name'])
  File "/home/skrenger/.local/lib/python3.10/site-packages/okd_camgi/interfaces.py", line 99, in pods
    for podname in os.listdir(pods_path):
FileNotFoundError: [Errno 2] No such file or directory: '/home/skrenger/workspace/cases/01234567/must-gather.local.817596368927932123/namespaces/openshift-machine-api/pods'
~~~

The issue is that `okd-camgi` is expecting the path to the inner directory in the `must-gather`, as the following will work as expected:

~~~
$ okd-camgi --webbrowser workspace/cases/01234567/must-gather.local.817596368927932123/quay-io-openshift-release-dev-ocp-v4-0-art-dev-sha256-415e56a6d8d2b8576c59d08a7c57a7d081fa59e7dc3cd8782ed7bcd6080c7c12/
file:///tmp/tmpg77gu4la/index.html
~~~

I believe it would be a good user experience if `okd-camgi` would try to see if the given path is valid and if there is a single subdirectory in there use that one.

There are some files present in all `must-gather` directories that you can use to more easily detect this (`version` in the inner directory and `timestamp` in the outer directory for example). Khizer Naeem has uses the following mechanism for `omg`: https://github.com/kxr/o-must-gather/blob/master/omg/cmd/use.py#L7

**Comment by elmiko** [comment]
thanks @simonkrenger , this is a great suggestion.

**Comment by elmiko** [comment]
i'm working on fixing this, but i am noticing that some of the must-gathers i have don't contain a `version` file in their root. i'm not sure what are the exact specifications for the must-gather and the enhancement is a little thin on details. i am going to proceed with the following rules:

1. look for the directories `namespaces` and `cluster-scoped-resources` in the specified path
2. if they exist, use that path
3. if they do not exist, look to see if there is a single directory in the current path, if so repeat step 1
4. if none of the above conditions are satisfied, return an error

**Comment by elmiko** [comment]
@simonkrenger i have released a new version `0.6.0` that includes a fix for this issue

**Comment by simonkrenger** [comment]
Great, thanks!
